toolchain/check/facet_type.h

```diff
@@ -60,14 +60,27 @@ auto AllocateFacetTypeImplWitness(Context& context,
                                   SemIR::InterfaceId interface_id,
                                   SemIR::InstBlockId witness_id) -> void;
 
-// Perform rewrite constraint resolution for a facet type and canonicalize the
-// FacetTypeInfo. The FacetTypeInfo must not be modified after this operation.
+// Perform rewrite constraint resolution for a facet type. The rewrite
+// constraints resolution is described here:
+// https://docs.carbon-lang.dev/docs/design/generics/appendix-rewrite-constraints.html#rewrite-constraint-resolution
 //
-// Returns false if an error was inserted into `facet_type`.
+// This function:
+// * Replaces the RHS of rewrite rules referring to `.Self` with the value
+//   coming from other rewrite rules. For example in `.X = () and .Y = .X` the
+//   result is `.X = () and .Y = ()`.
+// * Discards duplicate assignments to the same associated constant, such as in
+//   `.X = () and .X = ()` which becomes just `.X = ()`.
+// * Diagnoses multiple assignments of different values to the same associated
+//   constant such as `.X = () and .X = .Y`.
+// * Diagnoses cycles between rewrite rules such as `.X = .Y and .Y = .X` or
```

Comment:
Minor, but I think we're using `-` almost everywhere

```suggestion
// - Replaces the RHS of rewrite rules referring to `.Self` with the value
//   coming from other rewrite rules. For example in `.X = () and .Y = .X` the
//   result is `.X = () and .Y = ()`.
// - Discards duplicate assignments to the same associated constant, such as in
//   `.X = () and .X = ()` which becomes just `.X = ()`.
// - Diagnoses multiple assignments of different values to the same associated
//   constant such as `.X = () and .X = .Y`.
// - Diagnoses cycles between rewrite rules such as `.X = .Y and .Y = .X` or
```